### PR TITLE
[darwin] Allow Xcode 12 builds and iOS/tvOS SDK 14

### PR DESCRIPTION
**Pull request by sy6sy2**
## Description

Allow the use of Xcode 12 and iOS/tvOS SDK 14

## Motivation and Context

In order to be able to use Xcode 12 with latest iOS/tvOS SDKs.

## How Has This Been Tested?

Test performed on MacBook Pro 2015, macOS Catalina 10.15.6, Xcode 12

* tvOS target (SDK 14)
    * Build depends --> Ok
    * Build Kodi --> Ok
    * Run Kodi --> Ok

* iOS target (SDK 14)
    * Build depends --> Ok
    * Build Kodi --> Ok
    * Run Kodi --> Ok

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

**Comment by phunkyfish** [comment]
You would need build the deb files with XCode 12 to test this, you should also build depends.

I think for nettle at least you would require this patch: https://git.lysator.liu.se/nettle/nettle/-/merge_requests/9

File: tools/depends/target/nettle/Makefile [hunk 1]
```diff
@@ -22,6 +22,7 @@ $(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE) $(DEPS)
 	rm -rf $(PLATFORM)/*; mkdir -p $(PLATFORM)
 	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE)
 	cd $(PLATFORM); patch -p1 -i ../01-disable_testsuite.patch
+	cd $(PLATFORM); patch -p1 -i ../02-conftest_exit.patch
```
Comment (phunkyfish) [inline tools/depends/target/nettle/Makefile:25]: You also need to add this script to the DEPS at the top of this file

**Review by phunkyfish — CHANGES_REQUESTED** [review]

File: tools/depends/target/nettle/Makefile (hunk 1 above)
Comment (sy6sy2) [reply tools/depends/target/nettle/Makefile:25]: Done

**Comment by phunkyfish** [comment]
I can confirm that the nettle patch works for `inputstream.ffmpegdirect` too which is great. Tested on OSX.

**Comment by sy6sy2** [comment]
All tests are complete (see description).
Everything seems to be ok!

**Review by phunkyfish — APPROVED** [review]
👍

**Review by kambala-decapitator — APPROVED** [review]

**Comment by kambala-decapitator** [comment]
but I'm not sure why you tested macOS builds: you didn't enable macOS 11 SDK

**Comment by sy6sy2** [comment]
Because @phunkyfish asked me and I did not think xD
I am still with Catalina, I can't test SDK 11 anyway 😛

**Comment by kambala-decapitator** [comment]
you can't test features that require Big Sur but you can easily build against 11 SDK ;)

**Comment by sy6sy2** [comment]
Really?
But I not have SDK 11:

```
xcodebuild -showsdks
iOS SDKs:
	iOS 14.0                      	-sdk iphoneos14.0

iOS Simulator SDKs:
	Simulator - iOS 14.0          	-sdk iphonesimulator14.0

macOS SDKs:
	DriverKit 19.0                	-sdk driverkit.macosx19.0
	macOS 10.15                   	-sdk macosx10.15

tvOS SDKs:
	tvOS 14.0                     	-sdk appletvos14.0

tvOS Simulator SDKs:
	Simulator - tvOS 14.0         	-sdk appletvsimulator14.0

watchOS SDKs:
	watchOS 7.0                   	-sdk watchos7.0

watchOS Simulator SDKs:
	Simulator - watchOS 7.0       	-sdk watchsimulator7.0
```

**Comment by kambala-decapitator** [comment]
can't say why, but I have:
```
❯ xcodebuild -showsdks
iOS SDKs:
	iOS 14.0                      	-sdk iphoneos14.0

iOS Simulator SDKs:
	Simulator - iOS 14.0          	-sdk iphonesimulator14.0

macOS SDKs:
	DriverKit 20.0                	-sdk driverkit.macosx20.0
	macOS 11.0                    	-sdk macosx11.0

tvOS SDKs:
	tvOS 14.0                     	-sdk appletvos14.0

tvOS Simulator SDKs:
	Simulator - tvOS 14.0         	-sdk appletvsimulator14.0

watchOS SDKs:
	watchOS 7.0                   	-sdk watchos7.0

watchOS Simulator SDKs:
	Simulator - watchOS 7.0       	-sdk watchsimulator7.0
```

make sure that `xcode-select -p` outputs path to Xcode 12

**Comment by sy6sy2** [comment]
Strange... I have only one Xcode version. The one from Mac App Store.

```
> xcode-select -p
/Applications/Xcode.app/Contents/Developer
```

**Comment by kambala-decapitator** [comment]
download Xcode 12 beta - it should have 11 SDK.

I don't install from MAS and have a few Xcode versions :)

**Comment by sy6sy2** [comment]
Thank you, I will do!
Are you all ok to keep this PR open, then I can edit it in order to enable macOS 11 SDK (if it works of course)?

**Comment by kambala-decapitator** [comment]
I'm fine, but I suggest to split fixing build with Xcode 12 (the patch) and enabling new SDKs into separate commits

**Comment by phunkyfish** [comment]
@sy6sy2 if you add the Mac OS SDK I can test the build with it.

**Comment by phunkyfish** [comment]
No rush on this. Take your time.

**Review by fuzzard — COMMENTED** [review]
Imo it's wrong to patch configure
You should only patch aclocal.m4, then you should execute an autoreconf before the configure call
```
	cd $(PLATFORM); $(AUTORECONF)
```

**Comment by sy6sy2** [comment]
Where is this aclocal.m4 file.

According to similar old commits it sounds that we only need to modify configure.ac file (https://github.com/xbmc/xbmc/commit/22ae3fd53b9f22696dbb9bb5f0a9c2b120868bc2 and https://github.com/xbmc/xbmc/commit/9f6d891bb1bbce63f8047606af950c4f9c911428)

**Comment by fuzzard** [comment]
You have the aclocal.m4 file in your patch, you should drop the configure.ac portion of the patch file

**Comment by sy6sy2** [comment]
Oh, you talk about nettle! Sorry I was on the main configure.ac file -_-'
I will try! Thank you!

**Comment by wsnipex** [comment]
configure.ac is the input file for autoconf, so it is the correct file to patch

**Comment by fuzzard** [comment]
My concern is the upstream nettle patch only patches the M4 test, why patch something that doesn't look like will be upstreamed?

On top of the upstream pr already there, it conincides with what I had/was patching to get working a couple months back, go with whatever you want, but I have my doubts

**Comment by fuzzard** [comment]
Just noticed, typo on my end wsnipex the patch in this pr is against configure, and not configure.ac sorry the the confusion. Configure is the generated file, and therefore autoreconf will create the proper configure without the patching of it

**Comment by wsnipex** [comment]
I may be blind, but that patch is against configure.ac

**Comment by fuzzard** [comment]
https://github.com/xbmc/xbmc/pull/18451/files#diff-424ce9d0d7429a5029cee11fd1d2a3feR23-R25

nope, just configure

```
--- a/configure
+++ b/configure
@@ -4640,7 +4640,7 @@
```

**Comment by sy6sy2** [comment]
Yes Fuzzard is right, my nettle patch modify both `configure` and `aclocal.m4` files.
I have not the time ATM to try, but I guess Fuzzard is right : the patch only needs to modify `aclocal.m4` and we need to call `autoreconf` to ask a new `configure` file generation.
I will try this later today 😉 And also the macOS 11 SDK now that I have Xcode 12.2 beta with the missing SDK.

**Comment by sy6sy2** [comment]
* Nettle patch is now in a specific commit
* Nettle patch now only modify `aclocal.m4` file and not `configure` file anymore (xe use `autoreconf` command in order to generate the patched `configure` file)
* I tried macOS SDK 11 with Xcode 12.2 beta but the depends build fail on python3 --> So as Fuzzard suggests, we will only keep iOS/tvOS SDK ATM.

**Review by phunkyfish — APPROVED** [review]
Looks good

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by phunkyfish** [comment]
jenkins build this please

**Review by fuzzard — APPROVED** [review]

**Review by iAgentur — APPROVED** [review]

**Review by kambala-decapitator — APPROVED** [review]

**Pull request merged**